crates/core/src/kvs/ds.rs

```diff
@@ -617,14 +619,17 @@ impl Datastore {
 		let (builder, clock) = composer.new_transaction_builder(path, clock, canceller).await?;
 		//
 		let buckets = BucketsManager::new(Arc::new(composer));
+
+		let optimiser = Arc::new(Optimiser::all());
```

Comment:
The new --optimiser-max-passes CLI option is parsed and stored on EngineOptions, but the datastore always constructs the optimiser with Optimiser::all() using the environment-derived SURREAL_OPTIMISER_MAX_PASSES setting. The EngineOptions::optimiser_max_passes field is never consulted when building the optimiser, so the CLI flag currently has no effect. Consider threading the configured value into optimiser construction